cl8-web/src/store.js

```diff
@@ -386,17 +386,17 @@ export default new Vuex.Store({
     updateProfile: function(context, payload) {
       debug('sending update to Firebase', payload)
 
-      // not super happy about this - surely there's a toJSON() method?
+      // doing this round trip returns a JSON object we
+      // can save back to the realtime database more easily,
+      // and strips out properties we wouldn't want to save into it
       let newProfile = JSON.parse(JSON.stringify(payload))
```

Comment:
What is `newProfile` doing now, with `pushKey` now assigned to `payload['.key']`?

If it's getting set to something new below, can it just be assigned to undefined first?

Reply:
newProfile is acting as the updated JSON object that we want to save to firebase, with all the properties stripped out of the object that are added by something like Vue, and Vuefire.

The pushkey is the key used to work out where to put the updated user object in the userlist. It used to be in the JSON object we would store in the userlist, but it looks like it's not longer added by Vue, so we don't need to change it.

I guess we could make this a `const` now, as it doesn't change, and we don't need to remove the key anymore.

Does that help? I'm happy to do a quick call,  as this bit wasn't that intuitive when I first started working with it.

Reply:
I'm good with that. It wasn't clear to me with how the key was being assigned previously, but that all makes sense.